web/models/setting/system_variable.ex

```diff
@@ -49,6 +50,16 @@ defmodule CoursePlanner.SystemVariable do
     |> validate_editable()
   end
 
+  def validate_by_record_required(changeset, target_params) do
```

Comment:
I don't mean you should rename this function, I mean that in the `changeset(struct, params, :update)` in line 48 we call the Ecto validate_required function. Should we change it to call this one instead?